app/room_request/views.py

```diff
@@ -103,44 +103,49 @@ def delete_room_request(room_request_id):
     return redirect('/room-request/')
 
 
-@room_request.route('/<int:form_id>', methods=['GET', 'POST'])
-def viewID(form_id):
-    form = ActivityForm()
-    trasnferf = TransferForm()
-    user_found = False
-    comments = Activity.query.filter_by(room_request_id = form_id)
-    name = ""
-    try:
-        request_ob = RoomRequest.query.get(form_id)
-        print("User found")
-        name = str(request_ob.first_name) + " " + str(request_ob.last_name)
-        user_found = True
-    except: 
-        user_found = False
-    if form.validate_on_submit():
-            activity = Activity(
-                text=form.body.data,
-                commenter = current_user.first_name,
-                user_id = current_user.id,
-                room_request_id = request_ob.id
-                )
+@login_required
+@room_request.route('/<int:id>', methods=['GET', 'POST'])
+def view(id):    
+    room_request = RoomRequest.query.get(id)
+    if room_request is None:
+        return abort(404)
+
+    comments = Activity.query.filter_by(room_request_id=id)
+    activity_form = ActivityForm()
+    transfer_form = TransferForm()
+
+    if activity_form.validate_on_submit():
+        activity = Activity(
+            text=activity_form.body.data,
+            user_id=current_user.id,
+            room_request_id=room_request.id)
+        try:
             db.session.add(activity)
             db.session.commit()
             flash("Your comment has been added to the post", 'form-post')
-            form.body.data = ''
-            return render_template('room_request/id.html', id = form_id, name = name, user_found = user_found,
```

Comment:
Instead of including a `render_template` call inside each of the if-statement blocks, you can just call this at the end of the function! In this case, it's effectively the same thing, and makes the resulting code a bit cleaner :sparkles: